swarm/network/stream/delivery.go

```diff
@@ -42,17 +43,17 @@ var (
 )
 
 type Delivery struct {
```

Comment:
I think `Delivery` is seriously lacking documentation, also how the different parts play together - `Delivery`, SwarmChunkServer`, `SwarmChunkClient`, `ChunkStore`, etc.

I know it is a bit of a debt from the previous implementation but it is a good moment to add it.

Reply:
ok not sure in this pr though